### DISCUSSION TITLE
Add Config.anystr_upper and to_upper kwarg to constr and conbytes types

### DISCUSSION THREAD

**Discussion by candeira** [pydantic/pydantic#3976] [Feature Request]
This is not so much a feature request as a feature __proposal__: I'd like to add it myself.

Some codebases normalise case-independent strings to uppercase, and I'd like to add a `to_upper` option/normaliser to the `constr` and `conbytes` types, and also `Config.anystr_upper`.

**Comment by candeira** [comment]
I wanted to use this feature, so I went ahead and implemented it for myself. The PR is ready as soon as the feature is approved, all tests passing, docs updated, etc. Thanks so much for making it such a smooth process!

https://github.com/candeira/pydantic/tree/constr_to_upper

**Reply by samuelcolvin** [reply]
Thanks, I'll review.

**Reply by candeira** [reply]
Lovely, thanks!

**Comment by candeira** [comment]
Hey, @samuelcolvin! Bump! 

Is it ok if I open the PR?

**Comment by anonymouse-jj** [comment]
I'd like to have this feature as well

**Comment by wombatonfire** [comment]
Just for reference: `Config.anystr_upper` and `to_upper` for `constr`/`conbytes` were added in Pydantic 1.10.0.
